.github/workflows/docker-bases.yml

```diff
@@ -0,0 +1,90 @@
+name: Build base compilers docker images
+
+on:
+  push:
+    paths:
+      - './docker/Dockerfile.nvidia'
+      - './docker/Dockerfile.cpu'
+      - './docker/Dockerfile.amd'
+  workflow_dispatch:
+    inputs:
+      tags:
+        description: 'Build compiler bases'
+  schedule:
+    # Run once a week
+    - cron: "0 13 * * 1"
+
+jobs:
+  deploy-docker-bases:
+    name: ${{ matrix.tag }}
+    runs-on: ${{ matrix.runner }}
+    env:
+      DOCKER_BUILDKIT: "1"
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - tag: 'devitocodes/bases:nvidia-nvc'
+            arch: 'arch=nvc'
+            version: ''
+            dockerfile: './docker/Dockerfile.nvidia'
+            runner: ["self-hosted", "gpu", "openacc"]
+
+          - tag: 'devitocodes/bases:nvidia-clang'
+            arch: 'arch=clang'
+            version: 'ver=nvhpc-21-11'
+            dockerfile: './docker/Dockerfile.nvidia'
+            runner: ["self-hosted", "gpu", "openmp", "clang"]
+
+          - tag: 'devitocodes/bases:amd'
+            arch: ''
+            version: ''
+            dockerfile: './docker/Dockerfile.amd'
+            runner: ["self-hosted", "gpu", "docker", "aomp"]
+          
+          # These ones are tiny run on default
+          - tag: 'devitocodes/bases:cpu-gcc'
+            arch: 'arch=gcc'
+            version: ''
+            dockerfile: './docker/Dockerfile.cpu'
+            runner: ubuntu-latest 
+          
+          - tag: 'devitocodes/bases:cpu-icc'
```

Comment:
this hasn't been tested by CI yet right? worth performing a workflow dispatch before merge?

Reply:
scratch that, it is tested, I saw it now, but we need to make it "Required" on the github settings, will try now

Reply:
OK, done. 

Just noticed though that the icc-base build takes up more than 60 mins 🤯 
https://github.com/devitocodes/devito/runs/7243369665?check_suite_focus=true

Reply:
but the good news is that it passes all tests

Reply:
If I'm understanding this correctly, then if any one of three docker files is modified then all three are rebuilt? If so, then perhaps these should be 3 different actions?

Reply:
Interesting, I thought that this was not expected. Rerunning locally to check again.

Reply:
Well it's slightly more but all the CI take >=45min for base now because there is so much expensive gradient tests.